From 62c688ab3b22ae64d363a874a4f448323e2105a4 Mon Sep 17 00:00:00 2001
From: Leo Wang <leo.jt.wang@fii-foxconn.com>
Date: Thu, 29 May 2025 09:48:25 +0800
Subject: [PATCH 1003/1003] ARM dts aspeed clemente update DTS for DVT board

This commit introduces Device Tree (DTS) updates for the Clemente DVT
board, addressing I2C topology changes and adding initial support for
various components.

Key updates include:
- Reworked I2C aliases and device node definitions to align with
  the DVT board's I2C muxing and component placement. This affects
  EEPROMs, fan controllers, power monitors, and leak sensors.
- Adjusted I2C register addresses for several components based on
  DVT board specifications.
- Refined GPIO line names for main I/O expanders to accurately
  reflect their functions on the DVT hardware.
- Cleaned up obsolete I2C bus and IO expander definitions.

This enables fundamental hardware bringup and sensor monitoring
on the Clemente DVT platform.

Upstream-Status: Unknown
---
 .../aspeed/aspeed-bmc-facebook-clemente.dts   | 473 ++++++++++--------
 1 file changed, 268 insertions(+), 205 deletions(-)

diff --git a/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-clemente.dts b/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-clemente.dts
index a702f8641958..82a3f6f5b991 100644
--- a/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-clemente.dts
+++ b/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-clemente.dts
@@ -50,14 +50,14 @@ aliases {
 		i2c45 = &i2c0mux5ch1;
 		i2c46 = &i2c0mux5ch2;
 		i2c47 = &i2c0mux5ch3;
-		i2c48 = &i2c12mux0ch0;
-		i2c49 = &i2c12mux0ch1;
-		i2c50 = &i2c12mux0ch2;
-		i2c51 = &i2c12mux0ch3;
-		i2c52 = &i2c12mux0ch4;
-		i2c53 = &i2c12mux0ch5;
-		i2c54 = &i2c12mux0ch6;
-		i2c55 = &i2c12mux0ch7;
+		i2c48 = &i2c0mux0ch1mux0ch0;
+		i2c49 = &i2c0mux0ch1mux0ch1;
+		i2c50 = &i2c0mux0ch1mux0ch2;
+		i2c51 = &i2c0mux0ch1mux0ch3;
+		i2c52 = &i2c0mux3ch1mux0ch0;
+		i2c53 = &i2c0mux3ch1mux0ch1;
+		i2c54 = &i2c0mux3ch1mux0ch2;
+		i2c55 = &i2c0mux3ch1mux0ch3;
 	};
 
 	chosen {
@@ -235,6 +235,42 @@ i2c0mux0ch1: i2c@1 {
 			#address-cells = <1>;
 			#size-cells = <0>;
 			reg = <1>;
+
+			// HDD FRU EEPROM
+			eeprom@56 {
+				compatible = "atmel,24c128";
+				reg = <0x56>;
+			};
+
+			// E1.S Backplane
+			i2c0mux0ch1mux0: i2c-mux@74 {
+				compatible = "nxp,pca9546";
+				reg = <0x74>;
+				#address-cells = <1>;
+				#size-cells = <0>;
+				i2c-mux-idle-disconnect;
+
+				i2c0mux0ch1mux0ch0: i2c@0 {
+					#address-cells = <1>;
+					#size-cells = <0>;
+					reg = <0>;
+				};
+				i2c0mux0ch1mux0ch1: i2c@1 {
+					#address-cells = <1>;
+					#size-cells = <0>;
+					reg = <1>;
+				};
+				i2c0mux0ch1mux0ch2: i2c@2 {
+					#address-cells = <1>;
+					#size-cells = <0>;
+					reg = <2>;
+				};
+				i2c0mux0ch1mux0ch3: i2c@3 {
+					#address-cells = <1>;
+					#size-cells = <0>;
+					reg = <3>;
+				};
+			};
 		};
 		i2c0mux0ch2: i2c@2 {
 			#address-cells = <1>;
@@ -272,9 +308,9 @@ i2c0mux1ch1: i2c@1 {
 			reg = <1>;
 
 			// IO Mezz 0 IOEXP
-			io_expander7: gpio@21 {
+			io_expander7: gpio@20 {
 				compatible = "nxp,pca9535";
-				reg = <0x21>;
+				reg = <0x20>;
 				gpio-controller;
 				#gpio-cells = <2>;
 			};
@@ -284,6 +320,12 @@ eeprom@50 {
 				compatible = "atmel,24c64";
 				reg = <0x50>;
 			};
+
+			// OSFP 0 FRU EEPROM
+			eeprom@52 {
+				compatible = "atmel,24c128";
+				reg = <0x52>;
+			};
 		};
 		i2c0mux1ch2: i2c@2 {
 			#address-cells = <1>;
@@ -348,6 +390,42 @@ i2c0mux3ch1: i2c@1 {
 			#address-cells = <1>;
 			#size-cells = <0>;
 			reg = <1>;
+
+			// E1.S Backplane HDD FRU EEPROM
+			eeprom@56 {
+				compatible = "atmel,24c128";
+				reg = <0x56>;
+			};
+
+			// E1.S Backplane MUX
+			i2c0mux3ch1mux0: i2c-mux@74 {
+				compatible = "nxp,pca9546";
+				reg = <0x74>;
+				#address-cells = <1>;
+				#size-cells = <0>;
+				i2c-mux-idle-disconnect;
+
+				i2c0mux3ch1mux0ch0: i2c@0 {
+					#address-cells = <1>;
+					#size-cells = <0>;
+					reg = <0>;
+				};
+				i2c0mux3ch1mux0ch1: i2c@1 {
+					#address-cells = <1>;
+					#size-cells = <0>;
+					reg = <1>;
+				};
+				i2c0mux3ch1mux0ch2: i2c@2 {
+					#address-cells = <1>;
+					#size-cells = <0>;
+					reg = <2>;
+				};
+				i2c0mux3ch1mux0ch3: i2c@3 {
+					#address-cells = <1>;
+					#size-cells = <0>;
+					reg = <3>;
+				};
+			};
 		};
 		i2c0mux3ch2: i2c@2 {
 			#address-cells = <1>;
@@ -397,6 +475,12 @@ eeprom@50 {
 				compatible = "atmel,24c64";
 				reg = <0x50>;
 			};
+
+			// OSFP 1 FRU EEPROM
+			eeprom@52 {
+				compatible = "atmel,24c128";
+				reg = <0x52>;
+			};
 		};
 		i2c0mux4ch2: i2c@2 {
 			#address-cells = <1>;
@@ -443,43 +527,163 @@ i2c0mux5ch3: i2c@3 {
 &i2c1 {
 	status = "okay";
 
-	// HSC
+	// Module 0
+	fanctl0: fan-controller@20{
+		compatible = "maxim,max31790";
+		reg = <0x20>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+		channel@5 {
+			reg = <5>;
+			sensor-type = "TACH";
+		};
+		channel@6 {
+			reg = <6>;
+			sensor-type = "TACH";
+		};
+	};
+
+	// Module 0
+	fanctl1: fan-controller@23{
+		compatible = "maxim,max31790";
+		reg = <0x23>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+		channel@5 {
+			reg = <5>;
+			sensor-type = "TACH";
+		};
+		channel@6 {
+			reg = <6>;
+			sensor-type = "TACH";
+		};
+	};
+
+	// Module 1
+	fanctl2: fan-controller@2c{
+		compatible = "maxim,max31790";
+		reg = <0x2c>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+		channel@5 {
+			reg = <5>;
+			sensor-type = "TACH";
+		};
+		channel@6 {
+			reg = <6>;
+			sensor-type = "TACH";
+		};
+	};
+
+	// Module 1
+	fanctl3: fan-controller@27{
+		compatible = "maxim,max31790";
+		reg = <0x27>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+		channel@5 {
+			reg = <5>;
+			sensor-type = "TACH";
+		};
+		channel@6 {
+			reg = <6>;
+			sensor-type = "TACH";
+		};
+	};
+
+	// Module 0 Leak Sensor
+	adc@48 {
+        compatible = "ti,ads1015";
+        reg = <0x48>;
+        #address-cells = <1>;
+        #size-cells = <0>;
+
+        channel@0 {
+            reg = <0>;
+            ti,gain = <2>;
+        };
+
+        channel@1 {
+            reg = <1>;
+            ti,gain = <2>;
+        };
+
+        channel@2 {
+            reg = <2>;
+            ti,gain = <2>;
+        };
+
+        channel@3 {
+            reg = <3>;
+            ti,gain = <2>;
+        };
+    };
+
+	// Module 1 Leak Sensor
+	adc@49 {
+        compatible = "ti,ads1015";
+        reg = <0x49>;
+        #address-cells = <1>;
+        #size-cells = <0>;
+
+        channel@0 {
+            reg = <0>;
+            ti,gain = <2>;
+        };
+
+        channel@1 {
+            reg = <1>;
+            ti,gain = <2>;
+        };
+
+        channel@2 {
+            reg = <2>;
+            ti,gain = <2>;
+        };
+
+        channel@3 {
+            reg = <3>;
+            ti,gain = <2>;
+        };
+    };
+
+	// PDB
 	power-monitor@12 {
 		compatible = "ti,lm5066i";
 		reg = <0x12>;
 	};
 
-	// HSC
+	// PDB
 	power-monitor@14 {
 		compatible = "ti,lm5066i";
 		reg = <0x14>;
 	};
 
-	// HSC TEMP SENSOR
-	temperature-sensor@4a {
-		compatible = "ti,tmp461";
-		reg = <0x4a>;
+	// PDB TEMP SENSOR
+	temperature-sensor@4e {
+		compatible = "ti,tmp1075";
+		reg = <0x4e>;
 	};
 
-	// HSC FRU EEPROM
-	eeprom@53 {
+	// PDB FRU EEPROM
+	eeprom@50 {
 		compatible = "atmel,24c02";
-		reg = <0x53>;
+		reg = <0x50>;
 	};
 
-	// CPU0 Leak sensor MAX1363
-	Leak-sensor@34 {
-		compatible = "maxim,max1363";
-		reg = <0x34>;
+	// PDB
+	vrm@60 {
+		compatible = "renesas,raa228004";
+		reg = <0x60>;
 	};
 
-	// CPU1 Leak sensor MAX1363
-	Leak-sensor@35 {
-		compatible = "maxim,max1363";
-		reg = <0x35>;
+	// PDB
+	vrm@61 {
+		compatible = "renesas,raa228004";
+		reg = <0x61>;
 	};
 
-
+	// Interposer
 	i2c-mux@70 {
 		compatible = "nxp,pca9548";
 		#address-cells = <1>;
@@ -491,11 +695,6 @@ i2c1mux0ch0: i2c@0 {
 			#address-cells = <1>;
 			#size-cells = <0>;
 			reg = <0x0>;
-
-			power-sensor@22 {
-				compatible = "mps,mp5990";
-				reg = <0x22>;
-			};
 		};
 		i2c1mux0ch1: i2c@1 {
 			#address-cells = <1>;
@@ -506,35 +705,6 @@ i2c1mux0ch2: i2c@2 {
 			#address-cells = <1>;
 			#size-cells = <0>;
 			reg = <0x2>;
-
-			fanctl0: fan-controller@21{
-				compatible = "maxim,max31790";
-				reg = <0x21>;
-				#address-cells = <1>;
-				#size-cells = <0>;
-				channel@2 {
-					reg = <2>;
-					sensor-type = "TACH";
-				};
-				channel@5 {
-					reg = <5>;
-					sensor-type = "TACH";
-				};
-			};
-			fanctl1: fan-controller@27{
-				compatible = "maxim,max31790";
-				reg = <0x27>;
-				#address-cells = <1>;
-				#size-cells = <0>;
-				channel@2 {
-					reg = <2>;
-					sensor-type = "TACH";
-				};
-				channel@5 {
-					reg = <5>;
-					sensor-type = "TACH";
-				};
-			};
 		};
 		i2c1mux0ch3: i2c@3 {
 			#address-cells = <1>;
@@ -551,13 +721,13 @@ i2c1mux0ch5: i2c@5 {
 			#size-cells = <0>;
 			reg = <0x5>;
 
-			// PDB FRU EEPROM
-			eeprom@54 {
+			// Interposer FRU EEPROM
+			eeprom@51 {
 				compatible = "atmel,24c64";
-				reg = <0x54>;
+				reg = <0x51>;
 			};
 
-			// PDB TEMP SENSOR
+			// Interposer TEMP SENSOR
 			temperature-sensor@4f {
 				compatible = "ti,tmp75";
 				reg = <0x4f>;
@@ -568,27 +738,13 @@ i2c1mux0ch6: i2c@6 {
 			#size-cells = <0>;
 			reg = <0x6>;
 
-			// PDB IOEXP
+			// Interposer IOEXP
 			io_expander5: gpio@27 {
 				compatible = "nxp,pca9554";
 				reg = <0x27>;
 				gpio-controller;
 				#gpio-cells = <2>;
 			};
-
-			// OSFP IOEXP
-			io_expander6: gpio@25 {
-				compatible = "nxp,pca9555";
-				reg = <0x25>;
-				gpio-controller;
-				#gpio-cells = <2>;
-			};
-
-			// OSFP FRU EEPROM
-			eeprom@51 {
-				compatible = "atmel,24c64";
-				reg = <0x51>;
-			};
 		};
 		i2c1mux0ch7: i2c@7 {
 			#address-cells = <1>;
@@ -637,14 +793,6 @@ io_expander2: gpio@27 {
 		#gpio-cells = <2>;
 	};
 
-	// HMC Expander @0x74
-	io_expander3: gpio@74 {
-		compatible = "nxp,pca9555";
-		reg = <0x74>;
-		gpio-controller;
-		#gpio-cells = <2>;
-	};
-
 	// Module 0 Aux EEPROM
 	eeprom@50 {
 		compatible = "atmel,24c64";
@@ -660,7 +808,6 @@ eeprom@51 {
 
 &i2c3 {
 	status = "okay";
-
 };
 
 &i2c4 {
@@ -679,10 +826,11 @@ rtc@6f {
 		reg = <0x6f>;
 	};
 
-	// Interposer board EEPROM
-	eeprom@51 {
-		compatible = "atmel,24c64";
-		reg = <0x51>;
+	io_expander3: gpio@21 {
+		compatible = "nxp,pca9555";
+		reg = <0x21>;
+		gpio-controller;
+		#gpio-cells = <2>;
 	};
 };
 
@@ -764,81 +912,6 @@ ssif-bmc@10 {
 &i2c12 {
 	status = "okay";
 	multi-master;
-
-	i2c-mux@70 {
-		compatible = "nxp,pca9548";
-		reg = <0x70>;
-		#address-cells = <1>;
-		#size-cells = <0>;
-		i2c-mux-idle-disconnect;
-
-		i2c12mux0ch0: i2c@0 {
-			#address-cells = <1>;
-			#size-cells = <0>;
-			reg = <0>;
-		};
-		i2c12mux0ch1: i2c@1 {
-			#address-cells = <1>;
-			#size-cells = <0>;
-			reg = <1>;
-		};
-		i2c12mux0ch2: i2c@2 {
-			#address-cells = <1>;
-			#size-cells = <0>;
-			reg = <2>;
-		};
-		i2c12mux0ch3: i2c@3 {
-			#address-cells = <1>;
-			#size-cells = <0>;
-			reg = <3>;
-		};
-		i2c12mux0ch4: i2c@4 {
-			#address-cells = <1>;
-			#size-cells = <0>;
-			reg = <4>;
-		};
-		i2c12mux0ch5: i2c@5 {
-			#address-cells = <1>;
-			#size-cells = <0>;
-			reg = <5>;
-		};
-		i2c12mux0ch6: i2c@6 {
-			#address-cells = <1>;
-			#size-cells = <0>;
-			reg = <6>;
-			// HDD FRU EEPROM
-			eeprom@56 {
-				compatible = "atmel,24c64";
-				reg = <0x56>;
-			};
-		};
-		i2c12mux0ch7: i2c@7 {
-			#address-cells = <1>;
-			#size-cells = <0>;
-			reg = <7>;
-
-			power-sensor@40 {
-				compatible = "ti,ina230";
-				reg = <0x40>;
-				shunt-resistor = <2000>;
-			};
-			power-sensor@41 {
-				compatible = "ti,ina230";
-				reg = <0x41>;
-				shunt-resistor = <2000>;
-			};
-			power-sensor@44 {
-				compatible = "ti,ina230";
-				reg = <0x44>;
-				shunt-resistor = <2000>;
-			};
-			power-sensor@49 {
-				compatible = "ti,ina230";
-				reg = <0x49>;
-				shunt-resistor = <2000>;
-			};
-		};
-	};
 };
 
 &i2c13 {
@@ -1086,22 +1159,22 @@ &io_expander0 {
 
 &io_expander1 {
 	gpio-line-names =
-		"USB_MUX_SEL_N-O",
+		"SEC_FPGA_THERM_OVERT_L",
+		"SEC_FPGA_READY_BMC",
+		"SEC_HMC_BMC_DETECT",
+		"SEC_HMC_PGOOD",
 		"",
-		"MCIO_PCIE_DISABLE_L-O",
-		"USB_PCIE_DISABLE_L-O",
-		"PEX_SW_PCIE_DISABLE_L-O",
-		"HSC_CARD_PRSNT_N-I",
-		"TPM_MUX_3V3_SEL_OUT_N-O",
-		"USB2_HUB4_RST_L-O",
-		"PCI_MUX_SEL-O",
-		"TPM_MUX_SEL-O",
-		"IOEXP_BMC_PHY_RST_N-O",
-		"IOEXP_RTC_CLR_L-O",
-		"BMC_12V_CTRL-O",
-		"PS_RUN_IO0_PG-I",
+		"SEC_BMC_SELF_POWER_CYCLE",
+		"SEC_SEC_FPGA_EROT_FATAL_ERROR_L",
+		"SEC_WP_HW_EXT_CTRL_L",
+		"SEC_EROT_FPGA_RST_L",
+		"SEC_FPGA_EROT_RECOVERY_L",
+		"SEC_BMC_EROT_FPGA_SPI_MUX_SEL",
+		"SEC_USB2_HUB_RST_L",
 		"",
-		"";
+		"SEC_SGPIO_EN_L",
+		"SEC_IOB_IOEXP_INT_L",
+		"SEC_I2C_BUS_MUX_RESET_L";
 };
 
 &io_expander2 {
@@ -1126,18 +1199,18 @@ &io_expander2 {
 
 &io_expander3 {
 	gpio-line-names =
-		"IOB_PRSNT_L",
-		"IOB_DP_HPD",
-		"IOX_BMC_RESET",
-		"IOB_IOEXP_INT_L",
-		"IOB_UID_LED_L",
-		"IOB_UID_BTN_L",
-		"IOB_SYS_RST_BTN_L",
-		"IOB_PWR_LED_L",
-		"IOB_PWR_BTN_L",
-		"IOB_PHY_RST",
-		"CPLD_JTAG_MUX_SEL",
-		"",
+		"RTC_MUX_SEL",
+		"PCI_MUX_SEL",
+		"TPM_MUX_SEL",
+		"FAN_MUX-SEL",
+		"SGMII_MUX_SEL",
+		"DP_MUX_SEL",
+		"UPHY3_USB_SEL",
+		"NCSI_MUX_SEL",
+		"BMC_PHY_RST",
+		"RTC_CLR_L",
+		"BMC_12V_CTRL",
+		"PS_RUN_IO0_PG",
 		"",
 		"",
 		"",
@@ -1158,16 +1231,6 @@ &io_expander5 {
 		"","PSU_ALERT_N","","RST_P12V_STBY_N";
 };
 
-&io_expander6 {
-	gpio-line-names =
-		"OSFP_PHASE_ID0","OSFP_PHASE_ID1",
-		"OSFP_PHASE_ID2","OSFP_PHASE_ID3",
-		"","","","",
-		"OSFP_BOARD_ID0","OSFP_BOARD_ID1",
-		"OSFP_BOARD_ID2","PWRGD_P3V3_N1",
-		"PWRGD_P3V3_N2","","","";
-};
-
 &io_expander7 {
 	gpio-line-names =
 		"RST_CX7_0","RST_CX7_1",
-- 
2.49.0

